src/FarmCraft.Community.WebPortal/Services/FarmCraftApiService.cs

```diff
@@ -1,36 +1,85 @@
-﻿using FarmCraft.Community.WebPortal.Config;
+﻿using FarmCraft.Community.Data.DTOs;
+using FarmCraft.Community.Data.DTOs.Requests;
+using FarmCraft.Community.Services.Encryption;
+using FarmCraft.Community.WebPortal.Config;
 using Microsoft.Extensions.Options;
+using Newtonsoft.Json;
 using System.IdentityModel.Tokens.Jwt;
+using System.Text;
 
 namespace FarmCraft.Community.WebPortal.Services
 {
     public class FarmCraftApiService
     {
+        private readonly IEncryptionService _encryptor;
         private readonly HttpClient _httpClient;
+        private readonly ILogger<FarmCraftApiService> _logger;
         private JwtSecurityToken? _apiToken { get; set; }
```

Comment:
making it static won't work because the first logged-in user has their token used for everything. Leaving as-is but creating an issue to look for better options